.github/workflows/pregenerate.yml

```diff
@@ -0,0 +1,39 @@
+name: Pregenerate Files
+
+on:
+  pull_request:
+  push:
+    branches-ignore:
+      - main
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  update:
+    name: "Update"
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v3
+      - name: Fetch all history and metadata
+        run: |
+          git fetch --prune --unshallow
+          git checkout -b pr
+          git branch -f main origin/main
+      - name: Set up Python 3.9
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.9
+      - name: Install jinja
+        run: python -m pip install jinja2
+      - name: Configure committer identity
+        run: |
+          git config --global user.email "you@example.com"
+          git config --global user.name "Your Name"
```

Comment:
If a commit is never done, this isn't needed.
If a commit is done, this should be set to values indicating GH Actions and not placeholders.

Reply:
I'll admit I copied the basics from `.github/workflows/upstream-utils.yml`. I baulked at it too but figured there must have been a reason.

Removed.

Reply:
It's required for the upstream utils scripts to be able to apply patches.